### PR TITLE
logcli: add support for installing completion scripts for bash and zsh

### PR DESCRIPTION
**Pull request by remyleone**
Hello, 

I'm not sure about how to install the completion, so I tried to give it a shot looking at other formulas.
Here are how the scripts are crearted:

```
$ logcli --completion-script-bash

_logcli_bash_autocomplete() {
    local cur prev opts base
    COMPREPLY=()
    cur="${COMP_WORDS[COMP_CWORD]}"
    opts=$( ${COMP_WORDS[0]} --completion-bash ${COMP_WORDS[@]:1:$COMP_CWORD} )
    COMPREPLY=( $(compgen -W "${opts}" -- ${cur}) )
    return 0
}
complete -F _logcli_bash_autocomplete logcli
```

```
$ logcli --completion-script-zsh 

#compdef logcli
autoload -U compinit && compinit
autoload -U bashcompinit && bashcompinit

_logcli_bash_autocomplete() {
    local cur prev opts base
    COMPREPLY=()
    cur="${COMP_WORDS[COMP_CWORD]}"
    opts=$( ${COMP_WORDS[0]} --completion-bash ${COMP_WORDS[@]:1:$COMP_CWORD} )
    COMPREPLY=( $(compgen -W "${opts}" -- ${cur}) )
    return 0
}
complete -F _logcli_bash_autocomplete logcli
```

Is there a way to test that the completion was successfully installed?

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/l/logcli.rb [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# frozen_string_literal: true
+
+# Run LogQL queries against a Loki server
```
Comment (chenrui333) [inline Formula/l/logcli.rb:3]:
```suggestion
```

Comment (remyleone) [reply Formula/l/logcli.rb]: It was suggested by brew style. That's the reason why I added those lines.

Comment (ZhongRuoyu) [reply Formula/l/logcli.rb]: Did you run `brew style ./path/to/logcli.rb` where `logcli.rb` is outside the tap? For formulae we use `brew style logcli`.

**Review by ZhongRuoyu — APPROVED** [review]
Thanks!

> Is there a way to test that the completion was successfully installed?

The build would fail if it weren't.

**Pull request merged**